src/Components/Feature/ToDoList.tsx

```diff
@@ -2,41 +2,92 @@ import { useEffect, useState } from 'react';
 import { IToDoResp, IToDoRespD } from '../../Shared/types'
 import Todo from './ToDo'
 import useFetch from '../Common/useFetch';
+import FilterBar from './FIlterBar';
 
 
 const ToDoList = () => {
-  const [todos, setTodos] = useState<IToDoResp[] | null>(null);
-
+  const [originalTodos, setOriginalFilteredTodos] = useState<IToDoResp[]>([]);
+  const [filteredTodos, setFilteredTodos] = useState<IToDoResp[]>([]);
+  const [searchTerm, setSearchTerm] = useState<string>('');
+  const [sortBy, setSortBy] = useState<string>('');
+  const [statusFilter, setStatusFilter] = useState<string>('');
+  // const [todos, setTodos] = useState<IToDoResp[]>([]);
   const { data, loading, error }:IToDoRespD = useFetch('https://jsonplaceholder.typicode.com/todos');
 
   useEffect(() => {
     if (data) {
-      setTodos(data);
+      setFilteredTodos(data);
+      setOriginalFilteredTodos(data)
     }
   }, [data]);
 
   const markCompleted = (id: number) => {
-    if (todos) {
-      const updatedTodos:IToDoResp[] | null = todos.map(todo =>
+    if (filteredTodos) {
+      const updatedTodos:IToDoResp[] | null = filteredTodos.map(todo =>
         todo.id === id ? { ...todo, completed: !todo.completed } : todo
       );
-      setTodos(updatedTodos);
+      setFilteredTodos(updatedTodos);
     }
   };
 
   const deleteTodo = (id: number) => {
-    if (todos) {
-      const updatedTodos = todos.filter(todo => todo.id !== id);
-      setTodos(updatedTodos);
+    if (filteredTodos) {
+      const updatedTodos = filteredTodos.filter(todo => todo.id !== id);
+      setFilteredTodos(updatedTodos);
     }
   };
 
+    // Handle search input
+    const handleSearch = (value: string) => {
+      setSearchTerm(value);
+      filterTodos(value, sortBy, statusFilter);
+  };
+
+  // Handle sort dropdown
+  const handleSort = (value: string) => {
+      setSortBy(value);
+      filterTodos(searchTerm, value, statusFilter);
+  };
+
+  // Handle status dropdown
+  const handleStatus = (value: string) => {
+      setStatusFilter(value);
+      filterTodos(searchTerm, sortBy, value);
+  };
+
+  // Filter todos based on search term, sort, and status
+  const filterTodos = (search: string, sort: string, status: string) => {
+      let filtered = filteredTodos && filteredTodos.filter(todo =>
```

Comment:
try the optimal solution for the filter. Remove && and find alternative way to do this

Reply:
let filtered = filteredTodos ?? []

      if(!search){
        setFilteredTodos(originalTodos);
        filtered = originalTodos
      }
      else{
        filtered = filtered.filter(todo =>
          todo.title.toLowerCase().includes(search.toLowerCase())
        );
      }